hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/sink/StreamWriteOperatorCoordinator.java

```diff
@@ -408,8 +408,13 @@ private void initInstant(String instant) {
     HoodieTimeline completedTimeline = this.metaClient.getActiveTimeline().filterCompletedInstants();
     executor.execute(() -> {
       if (instant.equals(WriteMetadataEvent.BOOTSTRAP_INSTANT) || completedTimeline.containsInstant(instant)) {
-        // the last instant committed successfully
+        // We don't need to recommit for completed or failed writes
```

Comment:
The commit finishes, so the heartbeat should already been stopped?

Reply:
> The commit finishes, so the heartbeat should already been stopped?

Update and move  the logic before `startInstant`. Now the logic is, when bootstraping instant, if we do not recommit the instant and the instant is still inflight, we will stop the heartbeat for it.

Reply:
The incidence should be very low, let's keep the PR suspending to enlighten more ideas.